pytorch_lightning/trainer/trainer.py

```diff
@@ -792,44 +840,62 @@ def __test_using_best_weights(self, ckpt_path, test_dataloaders):
             model.load_state_dict(ckpt['state_dict'])
 
         # attach dataloaders
-        if test_dataloaders is not None:
-            self.data_connector.attach_dataloaders(model, test_dataloaders=test_dataloaders)
+        if dataloaders is not None:
+            kwargs = {'test_dataloaders' if stage == 'test' else 'val_dataloaders': dataloaders}
```

Comment:
{'test_dataloaders': dataloaders if stage == 'test' else 'val_dataloaders': dataloaders}

Reply:
This raises a SyntaxError on my machine (Python 3.8.6), I'm not sure if there's a way of doing that.

```
>>> dataloaders = []
>>> stage = 'test'
>>> {'test_dataloaders': dataloaders if stage == 'test' else 'val_dataloaders': dataloaders}
  File "<input>", line 1
    {'test_dataloaders': dataloaders if stage == 'test' else 'val_dataloaders': dataloaders}
                                                                              ^
SyntaxError: invalid syntax
```

Reply:
`{"a": 1 if False else 0}`
not
`{"a": 1 if False else "a": 0}`

Reply:
In this case it's the key that needs to change though, not the value.